specs/anchored-region/anchored-region.md

```diff
@@ -0,0 +1,280 @@
+        vertical-positioning-mode="onetime"
+    >
+      This shows up above or below the anchor depending on available space
+    </fast-anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that overlaps the anchor and renders above or below it depending on available space.
+
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <fast-anchored-region
+        anchor="anchor"
+        vertical-positioning-mode="dynamic"
+        vertical-inset="true"
+    >
+      This overlaps the anchor and extends above or below it depending on available space.
+    </fast-anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region renders above or below the anchor based on available space and is sized to match the available.
+
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <fast-anchored-region
+        anchor="anchor"
+        vertical-positioning-mode="dynamic"
+        vertical-scaling-enabled="true"
+    >
+       This region renders above or below the anchor based on available space and sizes itself to match.
+    </fast-anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that renders below the anchor until that space is less than 100px.
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <fast-anchored-region
+        anchor="anchor"
+        vertical-positioning-mode="dynamic"
+        vertical-default-position="below"
+        vertical-threshold="100"
+    >
+      This shows renders below the anchor as long as there is at least 100px available there.
+    </fast-anchored-region>
+    ...stuff...
+</div>
+```
+
+The dimensions of the anchored region will match the dimensions of the content unless scaling is enabled on a particular axis (verticalscalingenabled & horizontalscalingenabled) in which case it will fill all available space between the anchor and viewport.
+
+The component allows users to set a "Positioning Mode" on each axis which defines how the component will behave:
+- 'uncontrolled':  The anchored region will appear as it normally would in document flow.
+- 'locktodefault': The anchored region will always be placed in the specified default position regardless of available space.  For example a menu that always opens upwards.
+- 'dynamic': The anchored region is placed relative to the anchor element based on how much space is available between it and the viewport.  For example a menu that opens opens upwards when it is near the bottom of the page, and downwards when near the top.
+- 'onetime': As the 'dynamic' option, but position is not changed after initial placement.
+
+### API
+
+*Component name:*
+- `fast-anchored-region`
+
+*Attributes:*
+- anchor - The html id of the HTMLElement used as the anchor around which the positioning region is placed.  This must be set for the component's positioning logic to be active.
```

Comment:
For both `anchor` and `viewport` I think we should also have `anchorControl` and `viewportControl` properties (not attributes) which hold a direct reference to the anchor and viewport elements directly. When `anchor` or `viewport` changes, it will use the id to lookup the control and then set the appropriate control property. The rest can be driven off the control properties. This enables templates to actually directly bind to the control properties using refs and skip the id lookup if we want. So, you get an API that can work based off of id as well as control reference.

Reply:
Yes. I agree, will add that.

Reply:
Although I'm thinking anchorElement and viewportElement might be better than "Control"?

Reply:
I agree. I think the built-in label uses the "control" name but it seems more accurate to use "element" terminology.